src/Browscap/Data/DataCollection.php

```diff
@@ -224,6 +224,60 @@ public function addSourceFile($src)
     {
         $divisionData = $this->loadFile($src);
 
+        $this->validateDivisionDate($divisionData, $src);
+
+        if (isset($divisionData['versions']) && is_array($divisionData['versions'])) {
+            $versions = $divisionData['versions'];
+        } else {
+            $versions = ['0.0'];
+        }
+
+        if (isset($divisionData['userAgents']) && is_array($divisionData['userAgents'])) {
+            foreach ($divisionData['userAgents'] as $useragent) {
+                $this->validateUseragentData($useragent, $versions);
+
+                foreach ($useragent['children'] as $child) {
+                    if (!is_array($child)) {
+                        throw new \UnexpectedValueException(
+                            'each entry of the children property has to be an array for key "'
+                            . $useragent['userAgent'] . '"'
+                        );
+                    }
+
+                    $this->validateChildren($child, $useragent, $versions);
+                }
+
+                $this->allDivision[] = $useragent['userAgent'];
+            }
+
+            $userAgents = $divisionData['userAgents'];
+        } else {
+            $userAgents = [];
+        }
+
+        $this->divisions[] = new Division(
+            $divisionData['division'],
+            (int) $divisionData['sortIndex'],
+            $userAgents,
+            (boolean) $divisionData['lite'],
+            (boolean) $divisionData['standard'],
+            $versions,
+            substr($src, strpos($src, 'resources/'))
+        );
+
+        $this->divisionsHaveBeenSorted = false;
+
+        return $this;
+    }
+
+    /**
+     * @param array                        $divisionData
+     * @param string $src
+     *
+     * @throws \UnexpectedValueException
+     */
+    private function validateDivisionDate(array $divisionData, $src)
```

Comment:
Should this be named `validateDivisionData`?

Can `$src` be type-hinted as a `string`?